notebooks/ch16/classification_sgd.md

```diff
@@ -0,0 +1,187 @@
+
+
+```python
+# HIDDEN
+import numpy as np
+import matplotlib.pyplot as plt
+import pandas as pd
+import seaborn as sns
+%matplotlib inline
+import ipywidgets as widgets
+from ipywidgets import interact, interactive, fixed, interact_manual
+import nbinteract as nbi
+
+from sklearn.linear_model import SGDClassifier
+from sklearn.datasets import load_iris
+from sklearn.feature_extraction import DictVectorizer
+from sklearn.model_selection import train_test_split
+
+sns.set()
+sns.set_context('talk')
+np.set_printoptions(threshold=20, precision=2, suppress=True)
+pd.options.display.max_rows = 7
+pd.options.display.max_columns = 8
+pd.set_option('precision', 2)
+np.random.seed(42)
+```
+
+## Gradient Descent in Logistic Regression
+
+Previously, we covered batch gradient descent, an algorithm that iteratively updates $\hat\theta$ to find the cost-minimizing parameters. We also discussed stochastic gradient descent and mini-batch gradient descent, methods that take advantage of statistical theory and parallelized hardware to decrease the time spent training the gradient descent algorithm. In this section, we will apply these concepts specifically to logistic regression and walk through examples using scikit-learn functions.
```

Comment:
(Sorry for making comments on the .ipynb file for your Ch 11 section, it's probably easier to understand to the comments on .md files so I'll switch to that here).

nit: Replace "Previously" with "In chapter 11"

Reply:
Sam said we're trying not to reference chapter numbers, since they might be changed around in future versions of the textbook (case in point: classification just got moved from ch16 to ch17)